streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1220,6 +1269,9 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // generate producer configs from original properties and overridden maps
         final Map<String, Object> props = new HashMap<>(eosEnabled ? PRODUCER_EOS_OVERRIDES : PRODUCER_DEFAULT_OVERRIDES);
+        if (StreamThread.eosBetaEnabled(this) && !StreamThread.eosUpgradeModeEnabled(this)) {
+            props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, getString(StreamsConfig.APPLICATION_ID_CONFIG) + "-" + UUID.randomUUID());
```

Comment:
For eos-beta, we just generate a random UUID suffix for the TxId for the thread producer.

Reply:
Can we rely on `clientId`? From my understanding, `clientId` might be set to the same value for different `KafkaStreams` instances because quotas are based on `clientId`? For this case, we would get duplicate TxId? Thoughts?

Reply:
Good point.

Let's use the `processId` passed from `thread -> task-manager` then, since it is used in the subscription-info as well, and hence can also help with debugging as we link the values across different code paths.